phoenix-pherf/src/main/java/org/apache/phoenix/pherf/jmx/MonitorManager.java

```diff
@@ -0,0 +1,174 @@
+            } catch (InstanceAlreadyExistsException e) {
+                mbs.unregisterMBean(monitorThreadStatName);
+                mbs.registerMBean(bean, monitorThreadStatName);
+            }
+        }
+        this.rowCount = 0;
+        this.resultHandler = new CSVResultHandler(PherfConstants.MONITOR_FILE_NAME, ResultFileDetails.CSV);
+    }
+
+    @Override
+    public void run() {
+        try {
+            while (!shouldStop()) {
+                isRunning = true;
+                List rowValues = new ArrayList<String>();
+                synchronized (resultHandler) {
+                    for (MonitorDetails monitorDetails : MONITOR_DETAILS_LIST) {
+                        rowValues.clear();
+                        try {
+                            StandardMBean bean = new StandardMBean(monitorDetails.getMonitor(), Monitor.class);
+
+                            Calendar calendar = new GregorianCalendar();
+                            rowValues.add(monitorDetails);
+
+                            rowValues.add(((Monitor) bean.getImplementation()).getStat());
+                            rowValues.add(DateUtil.DEFAULT_MS_DATE_FORMATTER.format(calendar.getTime()));
+                            Result
+                                    result = new Result(ResultFileDetails.CSV, ResultFileDetails.CSV_MONITOR.getHeader().toString(), rowValues);
+                            resultHandler.write(result);
+                        } catch (Exception e) {
+                            throw new FileLoaderRuntimeException("Could not log monitor result.", e);
+                        }
+                        rowCount++;
+                    }
+                    try {
+                        Thread.sleep(getMonitorFrequency());
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                }
+            }
+        } finally {
+            try {
+                isRunning = false;
+                if (resultHandler != null) {
+                    resultHandler.flush();
+                    resultHandler.close();
+
+                }
+            } catch (Exception e) {
+                throw new FileLoaderRuntimeException("Could not close monitor results.", e);
+            }
+        }
+
+    }
+
+    public long getMonitorFrequency() {
+        return monitorFrequency;
+    }
+
+    public synchronized boolean shouldStop() {
+        return shouldStop;
+    }
+
+    public synchronized void stop() {
+        this.shouldStop = true;
+    }
+
+    public synchronized long getRowCount() {
+        return rowCount;
+    }
+
+    public synchronized boolean isRunning() {
+        return isRunning;
+    }
+
+    /**
+     * This method should really only be used for testing
+     *
+     * @return List < {@link org.apache.phoenix.pherf.result.Result} >
+     * @throws IOException
+     */
+    public synchronized List<Result> readResults() throws Exception {
```

Comment:
In that case you probably don't need synchronized on public synchronized List<Result> readResults() throws Exception, since the only state you are mucking with in this method is resultHandler which is already thread safe.